src/cgns_internals.c

```diff
@@ -81,7 +81,7 @@ void *cgi_malloc(size_t cnt, size_t size)
 {
     void *buf = calloc(cnt, size);
     if (buf == NULL) {
-        cgi_error("calloc failed for %d values of size %d", cnt, size);
+        cgi_error("calloc failed for %lu values of size %lu", (unsigned long)cnt, (unsigned long)size);
```

Comment:
I have heard, but not verified, that the size of `unsigned long` on 64-bit windows is 32 bits where the size of `size_t` is 64 bits.  If that is the case, then this can result in a truncation and meaningless error message since it would be possible for a `cnt` or `size` to be truncated in the error message.

If we are going to make this change, then there should be some compile-time assert that `sizeof(unsigned long) >= sizeof(size_t)` so we don't truncate values with these casts...

Reply:
I try to solve the problem by using %zu instead